--- a/libgcab/gcab-file.c
+++ b/libgcab/gcab-file.c
@@ -508,6 +508,10 @@
         }
     }
 
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+    g_autoptr(GFile) gstdout = g_file_new_for_path ("/dev/stdout");
+    return G_OUTPUT_STREAM (g_file_append_to (gstdout, 0, cancellable, error));
+#else
     /* create parent directories */
     g_autoptr(GFile) parent = g_file_get_parent (file);
     g_autoptr(GError) error_local = NULL;
@@ -522,6 +526,7 @@
     return G_OUTPUT_STREAM (g_file_replace (file, NULL, FALSE,
                                             G_FILE_CREATE_REPLACE_DESTINATION,
                                             cancellable, error));
+#endif
 }
 
 /**
