--- a/bzip2.c
+++ b/bzip2.c
@@ -1958,6 +1958,10 @@ IntNative main ( IntNative argc, Char *argv[] )
 #     endif
    }
 
+#ifdef __AFL_HAVE_MANUAL_CONTROL
+  __AFL_INIT();
+#endif
+
    if (opMode == OM_Z) {
      if (srcMode == SM_I2O) {
         compress ( NULL );
--- a/bzip2recover.c
+++ b/bzip2recover.c
@@ -363,6 +363,10 @@ Int32 main ( Int32 argc, Char** argv )
 
    strcpy ( inFileName, argv[1] );
 
+#ifdef __AFL_HAVE_MANUAL_CONTROL
+  __AFL_INIT();
+#endif
+
    inFile = fopen ( inFileName, "rb" );
    if (inFile == NULL) {
       fprintf ( stderr, "%s: can't read `%s'\n", progName, inFileName );
