--- a/libarchive_fe/passphrase.c
+++ b/libarchive_fe/passphrase.c
@@ -165,6 +165,9 @@
 static char *
 readpassphrase(const char *prompt, char *buf, size_t bufsiz, int flags)
 {
+	errno = EIO;
+	return(NULL);
+#if 0
 	ssize_t nr;
 	int input, output, save_errno, i, need_restart;
 	char ch, *p, *end;
@@ -302,6 +305,7 @@
 	if (save_errno)
 		errno = save_errno;
 	return(nr == -1 ? NULL : buf);
+#endif
 }
 #endif /* _WIN32 && !__CYGWIN__ */
 #endif /* HAVE_READPASSPHRASE */
